src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -1722,6 +1722,7 @@ private static void handlePutBlob(HttpServletRequest request,
         } catch (HttpResponseException hre) {
             HttpResponse hr = hre.getResponse();
             if (hr == null) {
+                response.sendError(HttpServletResponse.SC_BAD_GATEWAY);
```

Comment:
Maybe we should log `hre` to discover the cause?  Also perhaps `INTERNAL_ERROR` is more appropriate until we understand the root cause?